EcoSoapBank/Helpers/UIElementInitializers.swift

```diff
@@ -0,0 +1,39 @@
+//
+//  UIElementInitializers.swift
+//  EcoSoapBank
+//
+//  Created by Christopher Devito on 9/1/20.
+//  Copyright © 2020 Spencer Curtis. All rights reserved.
+//
+
+import UIKit
+
+class UIElementInitializers: NSObject {
+
+    /**
+     Create a UILabel with title, frame,alignment, and textColor passed in.
+
+     - Parameter title: **String** that describes the UILabel's text property
+     - Parameter frame: **CGRect** that describes the initial dimensions of the UILabel
+     - Parameter alignment: **NSTextAlignment** that describes the UILabel's  textAlignment property
+     - Parameter textColor: **UIColor** that describes the UILabel's textColor property
+     - Returns: A **UILabel** set up to the specifications passed in.
+     */
+    func createLabel(_ title: String, frame: CGRect, alignment: NSTextAlignment, textColor: UIColor = .label) -> UILabel {
+        let label = UILabel(frame: frame)
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.text = title
+        label.textAlignment = alignment
+        label.textColor = textColor
+
+        return label
+    }
+
+    func createImageView(_ imageNamed: String) -> UIImageView {
+        let imageView = UIImageView(image: UIImage(systemName: imageNamed, withConfiguration: UIImage.SymbolConfiguration(pointSize: 16, weight: .semibold)))
+        imageView.translatesAutoresizingMaskIntoConstraints = false
+        imageView.tintColor = .lightGray
+
+        return imageView
+    }
+}
```

Comment:
This is interesting! Definitely works and separates stuff out a bit.

Alternatively, it might be even simpler to make some convenience initializers on UILabel and UIImageView, something like this:

```swift
extension UILabel {
    convenience init(_ title: String, frame: CGRect, alignment: NSTextAlignment, textColor: UIColor = .label) {
        self.init(frame: frame)
        self.translatesAutoresizingMaskIntoConstraints = false
        self.text = title
        self.textAlignment = alignment
        self.textColor = textColor
    }
}

extension UIImageView {
    convenience init(customImageNamed name: String) {
        self.init(image: UIImage(systemName: imageNamed, withConfiguration: UIImage.SymbolConfiguration(pointSize: 16, weight: .semibold)))
        self.translatesAutoresizingMaskIntoConstraints = false
        self.tintColor = .lightGray
    }
}
```

That would avoid the need for a separate class. Though if you wanted to swap out one version of the class for another in various parts of the app or something, I could see that being handy.

Reply:
Hmm, fair point. Extensions to create convenience inits probably would be just as good. Either way, the main point was just to not have to write all this multiple times. Although, in the end I only called the image view one once so far and kinda made it too unique anyway, now that I look at it again (i.e. the .lightgray tint wouldn't work for any other imageviews).